### PR TITLE
Fix for OSFUSE-357: Do better PATH/executable detection on windows.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java [hunk 1]
```diff
@@ -123,6 +127,14 @@ protected static void processOutput(Logger log, InputStream inputStream, Functio
...
         List<File> pathDirectories = getPathDirectories(log);
         for (File directory : pathDirectories) {
+            if( isWindows() ) {
+                for (String extension : new String[]{".exe", ".bat", ".cmd"}) {
+                    File file = new File(directory, name+extension);
+                    if (file.exists() && file.isFile()) {
+                        return file;
+                    }
+                }
+            }
```
Comment (rhuss) [inline core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java]: Shouldn't there be an `else` ?

File: core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java [hunk 2]
```diff
@@ -156,7 +168,14 @@ private static String canonicalPath(File file) {
 
     private static List<File> getPathDirectories(Logger log) {
         List<File> pathDirectories = new ArrayList<>();
-        String pathText = System.getenv("PATH");
+        HashMap<String, String> envs = new HashMap<>(System.getenv());
+        if( isWindows() ) {
+            // On windows, the PATH env var is case insensitive, force to upper case.
+            for (Map.Entry<String, String> entry : System.getenv().entrySet()) {
```
Comment (rhuss) [inline core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java]: why convert the whole environment when only the path is needed ?

```java
String getPath() {
  ...
  if (entry.getKey().toUpperCase().equals("PATH") {
      return entry.getValue()
  }
}
```

File: core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java [hunk 3]
```diff
@@ -123,6 +127,14 @@ protected static void processOutput(Logger log, InputStream inputStream, Functio
     public static File findExecutable(Logger log, String name) {
         List<File> pathDirectories = getPathDirectories(log);
         for (File directory : pathDirectories) {
+            if( isWindows() ) {
```
Comment (apupier) [inline core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java]: I would prefer to search first on the exact name and then with the extra extensions.

File: core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java [hunk 4]
```diff
@@ -123,6 +127,14 @@ protected static void processOutput(Logger log, InputStream inputStream, Functio
...
+                for (String extension : new String[]{".exe", ".bat", ".cmd"}) {
+                    File file = new File(directory, name+extension);
+                    if (file.exists() && file.isFile()) {
```
Comment (apupier) [inline core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java]: the "executability" is not checked for Windows

File: core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java (hunk 3 above)
Comment (apupier) [inline core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java]: you're right

seems that the best would be to be portable and use a java library

File: core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java (hunk 4 above)
Comment (apupier) [inline core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java]: image
there is a "read & execute" permission so I suppose that it is the one used

File: core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java (hunk 1 above)
Comment (rhuss) [inline core/src/main/java/io/fabric8/maven/core/util/ProcessUtil.java]: yes, makes sense.

**Comment by rhuss** [comment]
@chirino could you please rebase so that we can merge it ?

**Comment by chirino** [comment]
rebased and squashed.

**Comment by rhuss** [comment]
lgtm [merge]

**Pull request merged**